api/python/provisioner/commands/check.py

```diff
@@ -455,79 +457,18 @@ def _bmc_accessibility(*, args: str) -> CheckEntry:
-            res.set_fail(checked_target=cfg.ALL_MINIONS,
-                         comment="Package cortx-py-utils not installed")
-            return res
-
-        try:
-            # Get node list
-            nodes = Check._get_pillar_data("cluster/node_list")
-            # Check if BMC is accessible for these nodes
-            BmcV().validate('accessible', nodes)
-        except Exception as exc:
-            res.set_fail(checked_target=cfg.ALL_MINIONS,
-                         comment=str(exc))
-        else:
-            res.set_passed(checked_target=cfg.ALL_MINIONS)
-
-        return res
+        return Check._validate_bmc_check('accessible')
 
     @staticmethod
     def _bmc_stonith(*, args: str) -> Union[CheckEntry, List[CheckEntry]]:
         """
         Check BMC Stonith Configuration
 
-        :param args: bmc stonith configuration check specific parameters and arguments
+        :param args: bmc stonith configuration check specific
+                     parameters and arguments
         :return:
         """
-        check_entry: CheckEntry = CheckEntry(cfg.Checks.BMC_STONITH.value)
-        # Check for import errors
-        if cortx_py_utils_import_error:
-            check_entry.set_fail(checked_target=cfg.ALL_MINIONS,
-                                 comment="Package cortx-py-utils not installed")
-            return check_entry
-
-        try:
-            # Get node list
-            nodes = Check._get_pillar_data("cluster/node_list")
-        except Exception as exc:
-            check_entry.set_fail(checked_target=cfg.ALL_MINIONS,
-                                 comment=str(exc))
-            return check_entry
-
-        res: List[CheckEntry] = list()
-        _DECRYPT_PSWD_CMD = ('salt-call lyveutil.decrypt cluster {val}'
-                             ' --output=newline_values_only')
-
-        for node in nodes:
-            check_stonith: CheckEntry = CheckEntry(cfg.Checks.BMC_STONITH.value)
-
-            try:
-                # Get BMC IP, user and secret.
-                bmc_ip = Check._get_pillar_data(f"cluster/{node}/bmc/ip")
-                bmc_user = Check._get_pillar_data(f"cluster/{node}/bmc/user")
-                secret = Check._get_pillar_data(f"cluster/{node}/bmc/secret")
-
-                # Decrypt the secret
-                cmd = _DECRYPT_PSWD_CMD.format(val=secret)
-                minion_id = local_minion_id()
-                bmc_passwd = cmd_run(cmd, targets=minion_id)[minion_id]
-
-                # Check the stonith configuration
-                BmcV().validate('stonith',
-                                [node, bmc_ip, bmc_user, bmc_passwd])
-            except Exception as exc:
-                check_stonith.set_fail(checked_target=node, comment=str(exc))
-            else:
-                check_stonith.set_passed(checked_target=node)
-
-            res.append(check_stonith)
-
-        return res
+        return Check._validate_bmc_check('stonith')
```

Comment:
Same here. Make `_bmc_stonith` as bound class method by removing `@staticmethod` to refer to class method using `self.`:
```python
 return self._validate_bmc_check('stonith')
```

Reply:
comment is addressed